e2e/app/element/base-element.ts

```diff
@@ -31,15 +31,17 @@ export default class BaseElement extends Container {
    */
   async waitForXPath(waitOptions: WaitForSelectorOptions = { visible: true }): Promise<ElementHandle> {
     if (this.xpath === undefined && this.element !== undefined) return this.element.asElement();
-    try {
-      return this.page.waitForXPath(this.xpath, waitOptions).then((elemt) => (this.element = elemt.asElement()));
-    } catch (err) {
-      logger.error(`waitForXpath('${this.xpath}') failed`);
-      logger.error(err);
-      // Debugging pause
-      // await jestPuppeteer.debug();
-      throw new Error(err);
-    }
+    return this.page
+      .waitForXPath(this.xpath, waitOptions)
+      .then((elemt) => (this.element = elemt.asElement()))
+      .catch((err) => {
+        logger.error(`waitForXpath('${this.xpath}') failed`);
```

Comment:
If there are more of these catch blocks in the code we could create a wrapper that could perform this behavior and be applied to functions throughout. If you want to try that approach out let me know and we can pair on it.

Reply:
There are more. I like the idea of pairing. I'll set up a meeting next week. Feel free to reschedule if time doesn't work.